app/app.go

```diff
@@ -111,6 +112,10 @@ type App struct {
 	// UUID is a v4 UUID lazily generated on the first call to GetUUID()
 	UUID string
 
+	// InterApp Properties implemented by provision.InterAppProvisioner
+	// it is lazy generated on the first call to FillInternalAddresses
+	InternalAddresses []*provision.AppInternalAddress `json:",omitempty" bson:"-"`
```

Comment:
This could be a `[]provision.AppInternalAddress` I don't think there's a need for pointers here.